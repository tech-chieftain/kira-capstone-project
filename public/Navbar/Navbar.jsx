import React from "react";

import Button from "react-bootstrap/Button";
import { getFreelancers } from "src/Utilities/FirebaseUtilities";
import { Header } from "./Navbar.styled";

import NavLogo from "./Logo";
import { Container, Btn } from "./Navbar.styled";

const Navbar = () => (
  <Container>
    <div className="logo-container">
      <NavLogo />
    </div>
    <Btn>
      <button type="button" className="join">
        Join
      </button>
      <button type="button" className="login" onClick={getFreelancers}>
        Log In
      </button>
    </Btn>
  </Container>

);

export default Navbar;
