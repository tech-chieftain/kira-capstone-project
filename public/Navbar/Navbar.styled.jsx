/* eslint-disable import/prefer-default-export */
import styled from "styled-components";

export const Container = styled.nav`
  background: #008080;
  height: 64px;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: space-between;
  position: fixed;
  top: 0%;
  left: 0%;

  .logo-container {
    margin-left: 105px;
  }
`;
export const Btn = styled.div`
  /* display: flex;
  align-items: center;
  justify-content: flex-end; */
  position: relative;
  right: 40px;
  margin: 0, 100px;
  margin-right: 105px;
  .join {
    height: 32px;
    width: 112px;
    color: #006767;
    background-color: white;
    border-radius: 5px;
    border: none;
  }
  .login {
    height: 32px;
    width: 112px;
    color: #fff;
    background-color: #008080;
    border-radius: 5px;
    border: none;
    margin-right: 50px;
  }
`;
