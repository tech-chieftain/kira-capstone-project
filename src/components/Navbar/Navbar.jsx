import React from "react";
import Button from "react-bootstrap/Button";
//import { Header } from "./Navbar.styled";

const Navbar = () => (
  <div>
    <p>This is an example of a component</p>
    <Button variant="primary">Example of Antd button</Button>
  
  </div>
);

export default Navbar;