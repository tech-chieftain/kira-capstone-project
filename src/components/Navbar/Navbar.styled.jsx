import styled from "styled-components";
import { Button, Image, Dropdown, Form } from "react-bootstrap";

export const JoinBtn = styled(Button)`
  height: 32px;
  width: 112px;
  color: #006767;
  background: #fff;
  border-radius: 5px;
  border: none;
`;

export const LoginBtn = styled(Button)`
  height: 32px;
  width: 112px;
  color: #fff;
  background: #008080;
  border-radius: 5px;
  border: none;
  margin-right: 75px;
`;

export const Img = styled(Image)`
  margin: -8px 0 0 -30px;

  @media (max-width: 1200px) {
    margin: 0;
  }
`;

export const ProfileImg = styled(Image)`
  width: 28px;
  height: 28px;
  overflow: hidden;
`;

export const DropDown = styled(Dropdown)`
  .dropdown-basic {
    color: white;
    box-shadow: none;
    webkit-box-shadow: none;
    margin: 3px auto 0;

    @media (max-width: 1200px) {
      margin-left: -11px;
    }
  }
`;

export const SearchForm = styled(Form)`
  margin-top: 3px;
  .input {
    border-radius: 0 5px 5px 0;
    margin-left: -10px;
    border: none;
    box-shadow: none;
    webkit-box-shadow: none;
    outline: none;
    width: 370px;
    height: 40px;
    &:focus {
      color: grey;
    }
  }

  #basic-addon1 {
    background: white;
    border-radius: 5px 0 0 5px;
    border: none;
    color: grey;
  }

  #dropdown-basic-button {
    background: transparent;
    color: grey;
    border: none;
    padding: 0 5px;
    box-shadow: none;
  }
`;
