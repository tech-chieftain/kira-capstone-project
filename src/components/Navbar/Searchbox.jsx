import React, { useState } from "react";
import { useRouter } from "next/router";
import DropdownButton from "react-bootstrap/DropdownButton";
import Dropdown from "react-bootstrap/Dropdown";
import InputGroup from "react-bootstrap/InputGroup";
import { IoSearch } from "react-icons/io5";
import { useTranslation } from "next-i18next";
import { SearchForm } from "./Navbar.styled";

const Searchbox = () => {
  const [input, setInput] = useState("");
  const [searchForFreelancers, setSearchForFreelancers] = useState(true);
  const [searchPlaceholder, setSearchPlaceholder] = useState("navbar.findFreelancer");
  const setdropdowntrue = () => {
    setSearchForFreelancers(true);
    setSearchPlaceholder("navbar.findFreelancer");
  };
  const setdropdownfalse = () => {
    setSearchForFreelancers(false);
    setSearchPlaceholder("navbar.findService");
  };
  const router = useRouter();
  const handleSubmit = (e) => {
    e.preventDefault();
    const q = `/${searchForFreelancers ? "freelancers" : "services"}_search?q=${input}`;
    router.push(q);
  };
  const handleInput = (e) => {
    setInput(e.target.value);
  };

  const { t } = useTranslation("navbar");
  return (
    <>
      <SearchForm className="d-flex md-shadow mx-2" onSubmit={handleSubmit}>
        <InputGroup.Text id="basic-addon1">
          <IoSearch size="25px" />
        </InputGroup.Text>
        <input
          type="search"
          placeholder={t(searchPlaceholder)}
          aria-label="Search"
          className="input"
          onChange={handleInput}
        />
      </SearchForm>
      <DropdownButton id="dropdown-basic-button">
        <Dropdown.Item onClick={setdropdowntrue}>{t("navbar.freelancers")}</Dropdown.Item>
        <Dropdown.Item onClick={setdropdownfalse}>{t("navbar.services")}</Dropdown.Item>
      </DropdownButton>
    </>
  );
};
export default Searchbox;
