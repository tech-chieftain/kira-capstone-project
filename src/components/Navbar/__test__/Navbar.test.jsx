import * as React from "react";
import * as renderer from "react-test-renderer";
import Navbar from "../Navbar";

it("renders Navbar correctly", () => {
  const tree = renderer.create(<Navbar />).toJSON();
  expect(tree).toMatchSnapshot();
});
