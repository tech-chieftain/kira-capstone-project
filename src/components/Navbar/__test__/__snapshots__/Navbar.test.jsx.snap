// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Navbar correctly 1`] = `
<nav
  className="navbar navbar-expand-xl navbar-dark bg-primary"
>
  <div
    className="container"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      <img
        alt="Nabar logo"
        className="Navbarstyled__Img-sc-ua2yiz-2 kPMivA d-inline-block align-top"
        height="52px"
        src="/NavbarLogo.png"
        width="104px"
      />
    </a>
    <button
      aria-controls="responsive-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="responsive-navbar-nav"
    >
      <form
        className="Navbarstyled__SearchForm-sc-ua2yiz-5 logRsi d-flex md-shadow mx-2"
        onSubmit={[Function]}
      >
        <span
          className="input-group-text"
          id="basic-addon1"
        >
          <svg
            fill="currentColor"
            height="25px"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="25px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M456.69 421.39L362.6 327.3a173.81 173.81 0 0034.84-104.58C397.44 126.38 319.06 48 222.72 48S48 126.38 48 222.72s78.38 174.72 174.72 174.72A173.81 173.81 0 00327.3 362.6l94.09 94.09a25 25 0 0035.3-35.3zM97.92 222.72a124.8 124.8 0 11124.8 124.8 124.95 124.95 0 01-124.8-124.8z"
            />
          </svg>
          <div
            className="dropdown"
          >
            <button
              aria-expanded={false}
              aria-haspopup={true}
              className="dropdown-toggle btn btn-primary"
              disabled={false}
              id="dropdown-basic-button"
              onClick={[Function]}
              type="button"
            />
          </div>
        </span>
        <input
          aria-label="Search"
          className="input"
          onChange={[Function]}
          placeholder="navbar.findFreelancer"
          type="search"
        />
      </form>
      <div
        className="w-100 d-flex justify-content-end navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-link"
          data-rb-event-key="/add_service"
          href="/add_service"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          <svg
            className="text-white mt-1"
            fill="currentColor"
            height="25px"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 448 512"
            width="25px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
            />
          </svg>
        </a>
        <div
          className="Navbarstyled__DropDown-sc-ua2yiz-4 yjALK dropdown"
        >
          <button
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-basic dropdown-toggle btn btn-primary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <svg
              className="text-white"
              fill="currentColor"
              height="25px"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="25px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 288c79.5 0 144-64.5 144-144S335.5 0 256 0 112 64.5 112 144s64.5 144 144 144zm128 32h-55.1c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16H128C57.3 320 0 377.3 0 448v16c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48v-16c0-70.7-57.3-128-128-128z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</nav>
`;
